src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -423,12 +427,29 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         return this.getOrCreateControllerForActiveInterpreter(activeInterpreter, notebookType);
     }
     @traceDecoratorVerbose('Get default Remote Controller')
-    private async createDefaultPythonRemoteController(
-        notebookType: typeof JupyterNotebookView | typeof InteractiveWindowView
+    private async createDefaultRemoteController(
+        notebookType: typeof JupyterNotebookView | typeof InteractiveWindowView,
+        notebook?: NotebookDocument
     ) {
+        const metadata = notebook ? getNotebookMetadata(notebook) : undefined;
+        const language =
+            !metadata || isPythonNotebook(metadata) || !metadata.language_info?.name
+                ? PYTHON_LANGUAGE
+                : metadata.language_info.name;
+        const kernelName = metadata ? metadata.kernelspec?.name : undefined;
         // Get all remote kernels
         await this.loadNotebookControllers();
-        const controllers = this.getRegisteredNotebookControllers();
+        const controllers = this.getRegisteredNotebookControllers().filter((item) => {
+            // Sort out interactive or non-interactive controllers
+            if (
+                item.connection.kind !== 'startUsingRemoteKernelSpec' ||
+                (notebookType === 'jupyter-notebook' && item.id.includes(InteractiveControllerIdSuffix)) ||
```

Comment:
This logic seems backwards? notebooktype==='jupyter-notebook' would never have the interactive controller id suffix?